cloud/api.rst

```diff
@@ -0,0 +1,167 @@
+Formation API
+#############
+
+.. note::
+
+   Cloud API access is available upon request. Open a :ref:`support ticket <cloud_support>` to enable it for your account.
+
+Citus Cloud provides an API to list, delete, and create formations within an organization. This is a convenient way to manage frequent formation changes.
+
+Authentication
+==============
+
+All formation API requests must include an authentication token. To get one, visit the `tokens <https://console.citusdata.com/api_tokens>`_ page, accessible through the Cloud menu bar in ``Org > API Tokens``:
+
+.. image:: ../images/cloud-api-get-token.png
+
+Create a token with a name that's meaningful to you. The name does not affect the API, it exists so you can revoke the token later if desired. Tokens last for one year by default.
+
+.. image:: ../images/cloud-api-name-token.png
+
+After generating the token, its value will be displayed just once. Copy it to a secure place because it will never be visible again from the Cloud console.
+
+In the examples below, we will assume that the token is assigned to an environment variable in the shell:
+
+.. code-block:: bash
+
+   # save to an environment variable for convenience
+
+   export API_TOKEN=<token>
+
+The token is in `JWT <https://jwt.io/>`_ format, and should be added to HTTP requests using the ``Authorization`` header:
+
+.. code-block:: text
+
+   Authorization: Token API_TOKEN
+
+For security, all requests must be issued over https. Additionally, API requests are limited to five per minute.
+
+Create Formation
+================
+
+* Endpoint: POST https://console.citusdata.com/api/v1/formations/
+* Sample request payload:
+
+  .. code-block:: json
+
+    {
+       "name": "my-formation",
+       "region": "us-east-1",
+       "tune_attributes": {
+          "coordinator_tune": "t2.small",
+          "worker_tune": "t2.small",
+          "coordinator_ha": false,
+          "worker_ha": false,
+          "worker_count": "2",
+          "coordinator_disk_size": "100",
+          "worker_disk_size": "100"
+       }
+    }
+
+  Parameters:
+
+  * (mandatory) **name**: a non-empty string of your choice
+  * (mandatory) **region**: any of our :ref:`cloud_regions`
+  * (mandatory) **coordinator_tune**, **worker_tune**: a tune is any of these AWS machine types: t2.small, t2.medium, t2.large,  r3.large, r3.xlarge, r3.2xlarge, r3.4xlarge, r3.8xlarge r4.large, r4.xlarge, r4.2xlarge, r4.4xlarge, r4.8xlarge, r4.16xlarge, c4.xlarge, c4.2xlarge, c4.4xlarge, c4.8xlarge, c5.xlarge, c5.2xlarge, c5.4xlarge, c5.9xlarge. *(If you choose a tune from the t2.\* series, both the coordinator_tune and worker_tune must be same.)*
```

Comment:
We now also support; r5.large, r5.xlarge, r5.2xlarge, r5.4xlarge, r5.12xlarge, r5.24xlarge.